ipc/cli/src/commands/subnet/join.rs

```diff
@@ -29,7 +32,19 @@ impl CommandLineHandler for JoinSubnet {
             Some(address) => Some(require_fil_addr_from_str(address)?),
             None => None,
         };
-        let public_key = hex::decode(&arguments.public_key)?;
+        let keystore = provider.evm_wallet()?;
+        let address_str = arguments
+            .from
```

Comment:
`from` is an optinal argument, if its not provided should we not try to use the default address if its set?

Reply:
Thanks @fridrik01 ! I'll look into this and update the PR with a solution. I think that from was optional before this PR but because public-key was required 'from' was in essence required. I will update the PR with either

a) support to use the default address when `from` is not specified
b) Update `from` to be a required variable. 

I will go for a) and look into what the other cli options do when from is optional and use that

Reply:
This is beginning to look a bit more complicated that I had anticipated. The code that translates the optional input "From" is at the provider level and the current PR has changes in the "command" section of the code base which does not have access to the code to access the default "from" account. I think I will need to modify this PR to have the provider look up the public key in fund_with_token

Reply:
@fridrik01 fixed here https://github.com/consensus-shipyard/ipc/pull/879/files - the code for this PR looks even cleaner now!